src/memkind_memtier.c

```diff
@@ -743,6 +746,11 @@ memtier_builder_construct_memtier_memory(struct memtier_builder *builder)
 MEMKIND_EXPORT void memtier_delete_memtier_memory(struct memtier_memory *memory)
 {
     print_memtier_memory(memory);
+
+    if (memory->get_kind == memtier_policy_data_movement_get_kind) {
```

Comment:
I thought about doing this POC without support for memtier_builder... The interface is not trivial and we would probably need to know the exact required parameters later. Please check other comments, we can discuss it tomorrow